src/sentry/grouping/strategies/newstyle.py

```diff
@@ -557,17 +537,6 @@ def single_exception(exception, context, **meta):
                 ],
             )
 
-        if context["detect_synthetic_exception_types"] and _synthetic_exception_type_re.match(
-            exception.type
-        ):
-            # Do not update type component of system variant, such that regex
-            # can be continuously modified without unnecessarily creating new
-            # groups.
```

Comment:
would restore this comment into the above codeblock

Reply:
would definetly continue to document why we don't want to touch system variant, this is hard to understand if you don't know the code in save_aggregate

Reply:
Restored the comment: https://github.com/getsentry/sentry/pull/27403/files#diff-765101e1faf45dddc42ee634ea872031228b3072cd6a28d41c533c3aba7414d0R531-R533